Submission of your work: you are required to upload the Notebook
TASKS  
(4 Marks) Create three DataFrames: df_a, df_b, and df_c using the data provided above.


(4 Marks) Use the pd.concat() function to merge the three DataFrames into a single DataFrame named all_sales.


(4 Marks) Display the combined DataFrame and verify that it contains records from all three branches.


(4 Marks) Add a new column named Region and assign values as follows:


Branch A → “Nairobi”


Branch B → “Mombasa”


Branch C → “Kisumu”


4. (4 Marks) Print summary statistics (using describe()) for the combined dataset and write one line of interpretation about the average sales.


In [1]:
# Branch A sales data
data_a = {

    'Branch': ['A', 'A', 'A'],

    'Product': ['Laptop', 'Mouse', 'Keyboard'],

    'Sales': [120000, 5000, 7000]

}


In [2]:
# Branch B sales data

data_b = {

    'Branch': ['B', 'B', 'B'],

    'Product': ['Laptop', 'Mouse', 'Monitor'],

    'Sales': [135000, 6000, 30000]

}


In [3]:
# Branch C sales data

data_c = {

    'Branch': ['C', 'C', 'C'],

    'Product': ['Laptop', 'Keyboard', 'Headset'],

    'Sales': [140000, 8000, 9000]

}

In [4]:
import pandas as pd

In [5]:
data_a=pd.DataFrame(data_a)
data_a

Branch   Product   Sales
0      A    Laptop  120000
1      A     Mouse    5000
2      A  Keyboard    7000

In [6]:
data_b=pd.DataFrame(data_b)
data_b

Branch  Product   Sales
0      B   Laptop  135000
1      B    Mouse    6000
2      B  Monitor   30000

In [7]:
data_c=pd.DataFrame(data_a)
data_c

Branch   Product   Sales
0      A    Laptop  120000
1      A     Mouse    5000
2      A  Keyboard    7000

In [8]:
df=pd.concat([data_a, data_b, data_c])
df

Branch   Product   Sales
0      A    Laptop  120000
1      A     Mouse    5000
2      A  Keyboard    7000
0      B    Laptop  135000
1      B     Mouse    6000
2      B   Monitor   30000
0      A    Laptop  120000
1      A     Mouse    5000
2      A  Keyboard    7000

In [9]:
data_a.shape[0]+data_b.shape[0]+data_c.shape[0]==df.shape[0]

True

In [10]:
#Add a new column named Region and assign values as follows:

df['Region']=df['Branch'].map({'A': 'Nairobi', 'B':'Mombasa', 'C':'Kisumu'})

In [11]:
df

Branch   Product   Sales   Region
0      A    Laptop  120000  Nairobi
1      A     Mouse    5000  Nairobi
2      A  Keyboard    7000  Nairobi
0      B    Laptop  135000  Mombasa
1      B     Mouse    6000  Mombasa
2      B   Monitor   30000  Mombasa
0      A    Laptop  120000  Nairobi
1      A     Mouse    5000  Nairobi
2      A  Keyboard    7000  Nairobi

###Print summary statistics (using describe()) for the combined dataset and write one line of interpretation about the average sales.

In [12]:
df.describe()

Sales
count       9.000000
mean    48333.333333
std     58185.049626
min      5000.000000
25%      6000.000000
50%      7000.000000
75%    120000.000000
max    135000.000000

## the average sales for all the employees sis 48333.3, though the data is scewed